be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -166,6 +167,24 @@ void ColumnReader::get_row_ranges_by_zone_map(CondColumn* cond_column, RowRanges
     _calculate_row_ranges(page_indexes, row_ranges);
 }
 
+Status ColumnReader::get_dict_page_decoder(BinaryDictPageDecoder* binaryDictPageDecoder) {
+    if (_column_dict_page_decoder == nullptr) {
+        PagePointer pp = _meta.dict_page();
+        PageHandle ph;
+        RETURN_IF_ERROR(read_page(pp, &ph));
+
+        Slice dict_data = ph.data();
+
+        std::shared_ptr<BinaryPlainPageDecoder> dict_page_decoder(
+                new BinaryPlainPageDecoder(dict_data));
```

Comment:
To retain memory for dict page, you should also hold `PageHandle`, otherwise PageCache could evict the page referenced by dict_page_decoder.

Reply:
👌